queries/plugins/text_to_num.py

```diff
@@ -0,0 +1,573 @@
+
+TöluðTalaÞúsundir →
+    TöluðTalaUndirÞúsund? TöluðTalaÞúsund
+
+TöluðTala10Til19Hundruð →
+    TöluðTala10Til19 TöluðTalaHundrað
+
+# HUNDRUÐ
+
+TöluðTalaUndirÞúsund →
+    TöluðTalaHundruð TöluðTalaTugurOgEining?
+    | TöluðTalaHundruð TöluðTalaOgUndirHundrað?
+    | TöluðTalaUndirHundrað
+
+TöluðTalaOgUndirÞúsund →
+    "og" TöluðTalaHundruð
+    | TöluðTalaOgUndirHundrað
+
+TöluðTalaHundruð →
+    TöluðTala1Til9? TöluðTalaHundrað
+
+# UNDIR HUNDRAÐ
+
+TöluðTalaUndirHundrað →
+    TöluðTalaTugurOgEining
+    | TöluðTalaTugir
+    | TöluðTala10Til19
+    | TöluðTala1Til9
+
+TöluðTalaOgUndirHundrað →
+    "og" TöluðTalaTugir
+    | "og" TöluðTala10Til19
+    | "og" TöluðTala1Til9
+
+TöluðTalaTugurOgEining →
+    TöluðTalaTugir "og" TöluðTala1Til9
+
+###
+
+TöluðTala0 → 'núll'
+
+TöluðTala1Til9 →
+    'einn:to'
+    | 'tveir:to'
+    | 'þrír:to'
+    | 'fjórir:to'
+    | "fimm"
+    | "sex"
+    | "sjö"
+    | "átta"
+    | "níu"
+
+TöluðTala10Til19 →
+    "tíu"
+    | "ellefu"
+    | "tólf"
+    | "þrettán"
+    | "fjórtán"
+    | "fimmtán"
+    | "sextán"
+    | "sautján"
+    | "átján"
+    | "nítján"
+
+TöluðTalaTugir →
+    "tuttugu"
+    | "þrjátíu"
+    | "fjörutíu"
+    | "fimmtíu"
+    | "sextíu"
+    | "sjötíu"
+    | "áttatíu"
+    | "níutíu"
+
+TöluðTalaHundrað → 'hundrað'
+
+TöluðTalaÞúsund → 'þúsund:töl'
+
+TöluðTalaMilljón → 'milljón'
+
+TöluðTalaMilljarður →
+    'milljarður'
+    | 'miljarður'
+
+TöluðTalaBilljón → 'billjón'
+
+TöluðTalaBilljarður → 'billjarður'
+
+TöluðTalaTrilljón → 'trilljón'
+
+TöluðTalaTrilljarður → 'trilljarður'
+
+TöluðTalaKvaðrilljón → 'kvaðrilljón'
+
+TöluðTalaKvaðrilljarður → 'kvaðrilljarður'
+
+TöluðTalaKvintilljón → 'kvintilljón'
+
+TöluðTalaSextilljón → 'sextilljón'
+
+TöluðTalaSeptilljón → 'septilljón'
+
+TöluðTalaOktilljón → 'oktilljón'
+
+"""
+
+
+_NUMBERS = {
+    "núll": 0,
+    "einn": 1,
+    "tveir": 2,
+    "þrír": 3,
+    "fjórir": 4,
+    "fimm": 5,
+    "sex": 6,
+    "sjö": 7,
+    "átta": 8,
+    "níu": 9,
+    "tíu": 10,
+    "ellefu": 11,
+    "tólf": 12,
+    "þrettán": 13,
+    "fjórtán": 14,
+    "fimmtán": 15,
+    "sextán": 16,
+    "sautján": 17,
+    "seytján": 17,
+    "átján": 18,
+    "nítján": 19,
+    "tuttugu": 20,
+    "þrjátíu": 30,
+    "fjörutíu": 40,
+    "fimmtíu": 50,
+    "sextíu": 60,
+    "sjötíu": 70,
+    "áttatíu": 80,
+    "níutíu": 90,
+    "hundrað": 100,
+    "þúsund": 1000,
+    "milljón": 10 ** 6,
+    "miljarður": 10 ** 9,
+    "milljarður": 10 ** 9,
+    "billjón": 10 ** 12,
+    "billjarður": 10 ** 15,
+    "trilljón": 10 ** 18,
+    "trilljarður": 10 ** 21,
+    "kvaðrilljón": 10 ** 24,
+    # Following numbers not in BÍN
+    "kvaðrilljarður": 10 ** 27,
+    "kvintilljón": 10 ** 30,
+    "sextilljón": 10 ** 36,
+    "septilljón": 10 ** 42,
+    "oktilljón": 10 ** 48,
+}
+
+
+# (math.prod is only available for python 3.8+)
+def _prod(li: Sequence[int]) -> int:
+    """Take product of non-empty list."""
+    p = li[0]
```

Comment:
Cool Python one-liner: `return functools.reduce(operator.__mul__, li, 1)`  (requires import functools, import operator)

Reply:
Actually plain `operator.mul` works, the dunder underscores are not needed.